akka-http2-support/src/main/scala/akka/http/impl/engine/http2/FrameLogger.scala

```diff
@@ -20,6 +20,10 @@ import FrameEvent._
 private[http2] object FrameLogger {
   final val maxBytes = 16
 
+  def logFramesIfEnabled(shouldLog: Boolean): BidiFlow[FrameEvent, FrameEvent, FrameEvent, FrameEvent, NotUsed] =
+    if (shouldLog) bidi
+    else BidiFlow.identity
```

Comment:
Does this 'disappear' on materialization or does it have a run-time cost?

Reply:
It should, if it doesn't it would be a bug (though, for HTTP/2 I'm not too concerned about runtime costs currently as we haven't really optimized anything so far).